packages/nimble-blazor/NimbleBlazor/Components/NimbleTable.razor.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace NimbleBlazor;
+
+/// <summary>
+/// A table component
+/// </summary>
+/// <typeparam name="TData">Represents the type for a row of data in the table (an element of the <see cref="Data"/>).</typeparam>
+/// <remarks>The type represented by <see cref="TData"/> should not have any hierarchy. All aspects that can be serialized
+/// should be at the top-level.</remarks>
+public partial class NimbleTable<TData> : ComponentBase
+{
+    private ElementReference _table;
+    private bool _shouldRender = false;
+    private IEnumerable<TData> _data = Enumerable.Empty<TData>();
+    internal static string SetTableDataMethodName = "NimbleBlazor.Table.setData";
+
+    [Inject]
+    private IJSRuntime? JSRuntime { get; set; }
+
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the data for the table.
+    /// </summary>
+    [Parameter]
+    public IEnumerable<TData> Data
+    {
+        get
+        {
+            return _data;
+        }
+        set
+        {
+            _data = value;
+            _shouldRender = true;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a callback that's invoked when the data changes
+    /// </summary>
+    [Parameter]
+    public EventCallback<IEnumerable<TData>> DataChanged { get; set; }
```

Comment:
Do we actually want this yet? we haven't set up the plumping for data change events to be observed from the table. I think that would correspond more to table editing which is down the line.

Reply:
This is required for data binding.

Reply:
It's not needed for 1-way binding (the components demo also could just consume the table like `<NimbleTable Data="TableData"></NimbleTable>` and the table would still get and render new data), but since all the other controls use 2-way binding and we'll need it soon enough it's probably fine to keep it.